client/dashboard/sites/index.tsx

```diff
@@ -66,6 +83,40 @@ export default function Sites() {
 	const { data: filteredData, paginationInfo } = filterSortAndPaginate( sites ?? [], view, fields );
 	const [ isModalOpen, setIsModalOpen ] = useState( false );
 
+	const handleOnViewChange = ( nextView: View ) => {
+		if ( nextView.type === 'list' ) {
+			return;
+		}
+
+		const _defaultView = {
+			...defaultView,
+			...DEFAULT_LAYOUTS[ nextView.type as keyof typeof DEFAULT_LAYOUTS ],
+		};
+
+		navigate( {
+			search: {
+				...currentSearchParams,
+				view: Object.fromEntries(
+					Object.entries( nextView ).filter(
+						( [ key, value ] ) =>
+							CONFIGURABLE_VIEW_KEYS.includes( key ) &&
```

Comment:
Do we really want to put the persistable keys in the URL too? I thought maybe it would only do it for `CONFIGURABLE_VIEW_KEYS.subtract(PERSISTABLE_VIEW_KEYS)` keys or something.

It is nice that you can override the stored preference with an explicit query param.

Reply:
Yeah I'd prefer to keep them in the URL as it is 🙂